pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -2114,6 +2114,25 @@ func TestApplyTaskResultsToPipelineResults(t *testing.T) {
 		},
 		expectedResults: nil,
 		expectedError:   fmt.Errorf("invalid pipelineresults [pipeline-result-1], the referred results don't exist"),
+	}, {
+		description: "object-results-resultname-not-exist",
+		results: []v1beta1.PipelineResult{{
+			Name:  "pipeline-result-1",
+			Value: *v1beta1.NewArrayOrString("$(tasks.pt1.results.bar.key1)"),
+		}},
+		taskResults: map[string][]v1beta1.TaskRunResult{
+			"pt1": {
+				{
+					Name: "foo",
+					Value: *v1beta1.NewObject(map[string]string{
+						"key1": "val1",
+						"key2": "val2",
+					}),
+				},
+			},
+		},
+		expectedResults: nil,
+		expectedError:   fmt.Errorf("invalid pipelineresults [pipeline-result-1], the referred results don't exist"),
```

Comment:
Can you also add a test case where the task name doesn't exist?

Reply:
There is a case already covers that